subsys/net/lib/dns/mdns_responder.c

```diff
@@ -38,11 +38,11 @@ LOG_MODULE_REGISTER(net_mdns_responder, CONFIG_MDNS_RESPONDER_LOG_LEVEL);
 #define MDNS_TTL CONFIG_MDNS_RESPONDER_TTL /* In seconds */
 
 #if defined(CONFIG_NET_IPV4)
-static struct net_context *ipv4;
-static struct sockaddr_in local_addr4;
+static struct net_context *ipv4[CONFIG_NET_IF_MAX_IPV4_COUNT];
+static struct sockaddr_in local_addr4[CONFIG_NET_IF_MAX_IPV4_COUNT];
```

Comment:
Do we need multiple `local_addr4` copies? After `setup_ipv4_addr()` we'll end up with an array of the same addresses, and `net_ipv4_igmp_join()` creates a copy anyway.

Reply:
Indeed, good point, will fix.